src/common/file-helpers.c

```diff
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: GPL-2.0-only
+#include <sys/stat.h>
+#include "common/file-helpers.h"
+
+bool
+file_exists(const char *filename)
```

Comment:
Can't we open and check the return code? Using stat() always leads to TOCTOU race. The severity of the race ranges from "minor" to "hurr durr I've rooted your PC".

Reply:
Yes I think that would work. Just have to make the button_*_load function return a bool.

This thing is a new one on me. Do you get that toctou problem even if you stat and then just open for read as a normal user?

Reply:
That's the inherent problem of having two separate calls, you can't work around it. It's not gonna cause any bugs here, but somewhat later it would be tempting use this functions somewhere where it should not be used.

Reply:
Got it

Reply:
Done